catalyst-gateway/bin/src/service/api/documents/put_document/mod.rs

```diff
@@ -46,48 +46,19 @@ pub(crate) type AllResponses = WithErrorResponses<Responses>;
 
 /// # PUT `/document`
 pub(crate) async fn endpoint(doc_bytes: Vec<u8>) -> AllResponses {
-    match CatalystSignedDocument::try_from(doc_bytes.as_slice()) {
+    match doc_bytes.as_slice().try_into() {
         Ok(doc) => {
-            let authors = doc
-                .authors()
-                .into_iter()
-                .map(|kid| kid.to_string())
-                .collect();
-
-            let doc_meta_json = match serde_json::to_value(doc.doc_meta()) {
-                Ok(json) => json,
-                Err(e) => {
-                    return AllResponses::internal_error(&anyhow!(
-                        "Cannot decode document metadata into JSON, err: {e}"
-                    ))
-                },
-            };
-
-            let doc_body = SignedDocBody::new(
-                doc.doc_id().into(),
-                doc.doc_ver().into(),
-                doc.doc_type().into(),
-                authors,
-                Some(doc_meta_json),
-            );
+            if let Err(e) = catalyst_signed_doc::validator::validate(&doc, &DocProvider).await {
+                // means that something happened inside the `DocProvider`, some db error.
+                return AllResponses::handle_error(&e);
```

Comment:
does that error provide enough context?

Reply:
In any other cases we are returning problem report